File: code/13-Java2Python/python/tests/testsupport/stubborn_number_guesser.py
from domain.number_guesser import NumberGuesser
from valueobjects.interval import Interval


class StubbornNumberGuesser(NumberGuesser):
    """Broken test double for NumberGuesser.

    Always returns the same fixed guess, ignoring the interval it is given.
    Used to verify that GuessGame cannot be driven into an infinite loop by
    a misbehaving guesser.
    """

    def __init__(self, fixed_guess: int) -> None:
        self._fixed_guess = fixed_guess

    def next_guess(self, interval: Interval) -> int:
        return self._fixed_guess

File: code/12-JavaCodeLoop/JavaTDDLoop/src/test/java/com/example/testsupport/StubbornNumberGuesser.java
package com.example.testsupport;

import com.example.domain.NumberGuesser;
import com.example.valueobjects.Interval;

/**
 * Broken test double for {@code NumberGuesser}.
 *
 * <p>Always returns the same fixed guess, ignoring the interval it is given. Used to
 * verify that {@code GuessGame} cannot be driven into an infinite loop by a
 * misbehaving guesser.</p>
 */
public class StubbornNumberGuesser extends NumberGuesser {

    private final int fixedGuess;

    public StubbornNumberGuesser(int fixedGuess) {
        this.fixedGuess = fixedGuess;
    }

    @Override
    public int nextGuess(Interval interval) {
        return fixedGuess;
    }
}
